src/biogeochem/FATESFireNoDataMod.F90

```diff
@@ -0,0 +1,177 @@
+module FATESFireNoDataMod
+
+#include "shr_assert.h"
+
+  !-----------------------------------------------------------------------
+  ! !DESCRIPTION:
+  ! module for FATES when not obtaining fire inputs from data
+  !
+  ! !USES:
+  use shr_kind_mod, only: r8 => shr_kind_r8, CL => shr_kind_CL
+  use shr_log_mod, only: errmsg => shr_log_errMsg
+  use abortutils, only: endrun
+  use clm_varctl, only: iulog
+  use decompMod, only: bounds_type
+  use CNFireMethodMod, only: cnfire_method_type
+  use CNFireBaseMod, only: cnfire_base_type
+  !
+  implicit none
+  private
+  !
+  ! !PUBLIC TYPES:
+  public :: fates_fire_no_data_type
+  !
+  type, extends(cnfire_base_type) :: fates_fire_no_data_type
+      ! !PRIVATE MEMBER DATA:
+      real(r8), public, pointer :: lnfm24(:)  ! Daily avg lightning by grid cell (#/km2/hr)
+    contains
+      ! !PUBLIC MEMBER FUNCTIONS:
+      procedure, public :: InitAccBuffer  ! Initialize accumulation processes
+      procedure, public :: InitAccVars  ! Initialize accumulation variables
+      procedure, public :: UpdateAccVars  ! Update/extract accumulations vars
+
+  end type fates_fire_no_data_type
+
+  character(len=*), parameter, private :: sourcefile = __FILE__
+  !
+  ! !PRIVATE MEMBER DATA:
+  !-----------------------------------------------------------------------
+
+  interface fates_fire_no_data_type
+     ! initialize a new cnfire_base object
+     module procedure constructor
+  end interface fates_fire_no_data_type
+  !-----------------------------------------------------------------------
+
+contains
+
+  !------------------------------------------------------------------------
+  type(fates_fire_no_data_type) function constructor()
+    !
+    ! !DESCRIPTION:
+    ! Creates an object of type fates_fire_no_data_type
+    ! !ARGUMENTS:
+    constructor%need_lightning_and_popdens = .true.
+  end function constructor
+
+  !-----------------------------------------------------------------------
+  subroutine InitAccBuffer (this, bounds)
```

Comment:
See my response [here](https://github.com/ESCOMP/CTSM/pull/991#discussion_r415340963).

If the group agrees with you, then yes, and do we make them "empty" or remove them from the module entirely?

Reply:
If you do decide to keep lnfm24, a better way to do this would be to have these methods in the NoData class and then extend it with the DataClass. That way they can both share the same accumulation averaging methods without having to duplicate them.

Reply:
If you do decide to keep them, they need to be made empty, because they will be called, they just want do anything. The calling interface makes them required, but in the No-Data case they don't do anything.

Reply:
Made empty.